### PR TITLE
Date lib updates, added functions and deprecated old versions.

### PR DESCRIPTION
**Pull request by dnegma**
## Context
Added helper functions, deprecated old versions. Moving the more general functions in `blueprints.util.date` into here.

## Testing done
Moved unit tests, more to come

### CODE REVIEW

File: src/toolbelt/date.clj [hunk 1]
```diff
@@ -349,4 +495,78 @@
...
+  :args (s/cat :date (s/or :inst inst? :unix integer?)
+               :timezone (s/? string?)))
+
+
+;; =============================================================================
+;; Deprecated
+;; =============================================================================
+
+
+(defmacro defdeprecated
```
Comment (allenjprice) [inline src/toolbelt/date.clj:527]: 🆒 🆒 🆒

File: src/toolbelt/date.clj [hunk 2]
```diff
@@ -220,6 +308,64 @@
...
+  (let [[y m d h min sec mill] (mapv (fnil identity 0) [year month day hour minute second millisecond])]
+    (to-date (t/date-time y m d h min sec mill))))
+
+
+(defn to-unix-time
+  "Return the number of seconds after the Unix time
+  Arity 1: returns the number of seconds after Unix time.
+  Arity 2: returns the time in `unit` after Unix time: #{:millis :seconds}"
+  ([dt]
+   (to-unix-time dt :millis))
```
Comment (allenjprice) [inline src/toolbelt/date.clj:348]: docstring above leads me to believe this should be `:seconds`, not `:millis`

**Review by allenjprice — APPROVED** [review]
These changes look great!

Just one comment where there appears to be an inconsistency between docstring and implementation in one of the functions. Otherwise: 🚢 🚢 🚢

**Review by jalehman — APPROVED** [review]

**Review by humbamp123 — APPROVED** [review]

File: src/toolbelt/date.clj (hunk 2 above)
Comment (dnegma) [reply src/toolbelt/date.clj:348]: doc string is wrong, I'll update

**Pull request merged**